peer.go

```diff
@@ -1122,12 +1122,20 @@ func (p *peer) PingTime() int64 {
 }
 
 // queueMsg queues a new lnwire.Message to be eventually sent out on the
-// wire.
-func (p *peer) queueMsg(msg lnwire.Message, doneChan chan struct{}) {
+// wire. It returns an error if we failed queing the message. An error
+// is sent on errChan if the message fails being sent to the peer, or
+// nil otherwise.
+func (p *peer) queueMsg(msg lnwire.Message, errChan chan error) error {
 	select {
-	case p.outgoingQueue <- outgoinMsg{msg, doneChan}:
+	case p.outgoingQueue <- outgoinMsg{msg, errChan}:
+		return nil
 	case <-p.quit:
-		return
+		peerLog.Debugf("Peer shutting down, could not enqueue msg.")
+		err := fmt.Errorf("peer shutting down")
+		if errChan != nil {
```

Comment:
If we're returning the error directly, do we need to add this to the channel? Since the error indicates that message wasn't queued at all, a non-nil would signify that reading from the `errChan` isn't necessary.

Reply:
Yeh I don't think we need to return the error. Instead we can just send over the channel as that lets the caller process the error in an async manner. Either is needed, not both though IMO.